docs/workers.md

```diff
@@ -178,8 +178,10 @@ recommend the use of `systemd` where available: for information on setting up
 
 ### `synapse.app.generic_worker`
 
-This worker can handle API requests matching the following regular
-expressions:
+This worker can handle API requests matching the following regular expressions.
+`GET` requests will be handled directly while `POST` and `PUT` requests will be
```

Comment:
This isn't strictly true. Some `GET` requests are proxied and not all `POST` or `PUT` requests are. I'm not really sure that this is useful information, and not something that is actionable by server admins :shrug:

Reply:
This makes this almost impossible to document or explain what is happening. 😢

Reply:
Can't we say something to the effect of "These endpoints can be routed to any worker. If a worker is set up to handle a stream then additional endpoints may be required to route to said worker, refer to the `streams` section below for further information."? And then for each stream type list what endpoints should be routed to them?

Reply:
I've added some text like that, hopefully it is clearer now.

I'm *fairly* certain all the endpoints I documented can be sent to any worker and will be proxied properly. 👍